### PR TITLE
[doc]update the compilation.md

### PR DESCRIPTION
**Pull request by luzhijing**
## Proposed changes

Update the compilation.md, add the docker image version explain.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: docs/en/installing/compilation.md [hunk 1]
```diff
@@ -45,22 +45,22 @@ This document focuses on how to code Doris through source code.
...
 | apache/incubator-doris:build-env-1.1 | [ff0dd0d](https://github.com/apache/incubator-doris/commit/ff0dd0d2daa588f18b6db56f947e813a56d8ec81) or later | 0.10.x or later |
 | apache/incubator-doris:build-env-1.2 | [4ef5a8c](https://github.com/apache/incubator-doris/commit/4ef5a8c8560351d7fff7ff8fd51c4c7a75e006a8) or later | 0.12.x - 0.14.0 |
 | apache/incubator-doris:build-env-1.3.1 | [ad67dd3](https://github.com/apache/incubator-doris/commit/ad67dd34a04c1ca960cff38e5b335b30fc7d559f) or later | 0.14.x |
-| apache/incubator-doris:build-env-1.4.1 | [24d3861](https://github.com/apache/incubator-doris/commit/24d38614a0f21ed606462816a262c2e1d8273ace) or later | 0.15.x(releasing) |
-| apache/incubator-doris:build-env-1.4.2 | [a81f4da](https://github.com/apache/incubator-doris/commit/a81f4da4e461a54782a96433b746d07be89e6b54) or later | 0.15.x(releasing) |
+| apache/incubator-doris:build-env-1.4.2      | [a81f4da](https://github.com/apache/incubator-doris/commit/a81f4da4e461a54782a96433b746d07be89e6b54) or later | 0.15.0          |
```
Comment (EmmyMiao87) [inline docs/en/installing/compilation.md:56]: Remove this line.

File: docs/en/installing/compilation.md [hunk 2]
```diff
@@ -45,22 +45,22 @@ This document focuses on how to code Doris through source code.
...
+| apache/incubator-doris:build-env-for-0.15.0 | [a81f4da](https://github.com/apache/incubator-doris/commit/a81f4da4e461a54782a96433b746d07be89e6b54) or later | 0.15.0          |
```
Comment (EmmyMiao87) [inline docs/en/installing/compilation.md:57]:
```suggestion
| apache/incubator-doris:build-env-for-0.15.0 | [a81f4da](https://github.com/apache/incubator-doris/commit/a81f4da4e461a54782a96433b746d07be89e6b54) or later | 0.15.x          |
```

File: docs/en/installing/compilation.md [hunk 3]
```diff
@@ -45,22 +45,22 @@ This document focuses on how to code Doris through source code.
     apache/incubator-doris   build-env-1.4.2    49f68cecbc1a        4 days ago          3.76GB
     ```
 
-Note: For different versions of Doris, you need to download the corresponding mirror version.
+Note: For different versions of Doris, you need to download the corresponding mirror version. From Apache Doris 0.15.0 version, the docker image will keep same version number with Doris, You can use  ` apache/incubator-doris:build-env-1.4.2 `  or ` apache/incubator-doris:build-env-for-0.15.0 `   compile Apache Doris 0.15.0, they are identical.
```
Comment (EmmyMiao87) [inline docs/en/installing/compilation.md:48]:
```suggestion
Note: For different versions of Doris, you need to download the corresponding image version. From Apache Doris 0.15.x version, the docker image will keep same version number with Doris, You can use  ` apache/incubator-doris:build-env-for-0.15.0 `  compile Apache Doris 0.15.0.
```

File: docs/en/installing/compilation.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ This document focuses on how to code Doris through source code.
 
 1. Download Docker Mirror
 
-    `$ docker pull apache/incubator-doris:build-env-1.3.1`
+    `$ docker pull apache/incubator-doris:build-env-1.4.2`
```
Comment (EmmyMiao87) [inline docs/en/installing/compilation.md:38]:
```suggestion
    `$ docker pull apache/incubator-doris:build-env-for-0.15.0`
```

File: docs/en/installing/compilation.md [hunk 5]
```diff
@@ -45,22 +45,22 @@ This document focuses on how to code Doris through source code.
     apache/incubator-doris   build-env-1.4.2    49f68cecbc1a        4 days ago          3.76GB
```
Comment (EmmyMiao87) [inline docs/en/installing/compilation.md:45]:
```suggestion
    apache/incubator-doris   build-env-for-0.15.0    49f68cecbc1a        4 days ago          3.76GB
```

**Review by luzhijing — COMMENTED** [review]
Thanks for your suggestions !

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**